Ensure that the hub client proxy generator correctly processes both explicit and inferred generic type arguments

MattyLeslie opened the pull request:
Fix HubClientProxyGenerator to Handle Inferred Generic Type Arguments

**Issue:**
The `HubClientProxyGenerator` was not generating proxy code for methods using inferred generic type arguments. This was because the initial syntax node filter only considered methods with explicit generic types.

**Solution:**
Updated the syntax node filter to include both explicit and inferred generic type arguments by changing the filter condition to `SimpleNameSyntax`. This ensures that all relevant methods are processed for proxy generation.

Fixes #43510

github-actions[bot] committed:
Update .NET SDK

Update .NET SDK to version 9.0.100-preview.5.24269.5.

---
updated-dependencies:
- dependency-name: Microsoft.NET.Sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

MattyLeslie committed:
Merge pull request #9 from MattyLeslie/update-dotnet-sdk-9.0.100-preview.5.24269.5

Update .NET SDK to 9.0.100-preview.5.24269.5

MattyLeslie committed:
changing the condition to check for SimpleNameSyntax, which is a base class for both GenericNameSyntax and IdentifierNameSyntax

MattyLeslie committed:
By explicitly checking for SimpleNameSyntax, we ensure that both generic and inferred cases are handled.

MattyLeslie committed:
Removing extra line

MattyLeslie committed:
Fixing IsSyntaxTargetForGeneration signature

BrennanConroy commented on src/SignalR/clients/csharp/Client.SourceGenerator/src/HubClientProxyGenerator.Parser.cs:
```diff
@@ -251,10 +245,11 @@ internal SourceGenerationSpec Parse(ImmutableArray<MethodDeclarationSyntax> meth
+                        && memberAccess.Parent.ChildNodes().FirstOrDefault(x => x is ArgumentListSyntax) is
+                            ArgumentListSyntax
+                        { Arguments: { Count: 1 } } als)
+
```
Why the newline?

BrennanConroy reviewed (commented):
Could you please add test(s)?
https://github.com/dotnet/aspnetcore/blob/main/src/SignalR/clients/csharp/Client/test/UnitTests/HubClientProxyGeneratorTests.cs

MattyLeslie committed:
Removing extra line

MattyLeslie committed:
Merge branch 'dotnet:main' into ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments

MattyLeslie committed:
Merge branch 'ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments' of https://github.com/mattyleslie/aspnetcore into ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments

MattyLeslie committed:
Add tests for HubClientProxyGenerator to validate handling of inferred and explicit generic type arguments.

MattyLeslie committed:
Merge branch 'dotnet:main' into ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments

MattyLeslie commented:
![image](https://github.com/dotnet/aspnetcore/assets/67387027/f8c5d4e1-eb2f-46cb-972b-1b349a1bf91e)
![image](https://github.com/dotnet/aspnetcore/assets/67387027/a309efd0-b3b3-42e6-88ba-6bf7cf606ded)

Hey @BrennanConroy, I added tests to `HubClientProxyGeneratorTests`, however after opening `HubClientProxyGeneratorTests` & `HubServerProxyGeneratorTests` when building I get an error with `GetHubServer` as depicted in the screenshots, when switching to the `main` branch, I am still presented with the errors. I don't see anything I've changed to attribute this to. I've tried re-setting my local environment to no avail. I just wanted to reach out to you and ensure I'm not missing something. Any direction would be appreciated.

BrennanConroy commented:
I think VS has some weird behavior with source generators being referenced via project reference. Sometimes building via the command line first might help. But the issue in the PR looks like you just need to make `MyClient` public or internal.

MattyLeslie committed:
Making MyClient internal

MattyLeslie committed:
Merge branch 'ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments' of https://github.com/mattyleslie/aspnetcore into ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments

MattyLeslie committed:
Fixing indentations

MattyLeslie committed:
Merge branch 'dotnet:main' into ensure-that-the-HubClientProxyGenerator-correctly-processes-both-explicit-and-inferred-generic-type-arguments
